docs/Displaying a User Interface/Main Screen Templates/index.md

```diff
@@ -134,6 +134,294 @@ if (success) {
+
+@![iOS]
+|~
+```objc
+// CustomSDLScreen.h
+@class SDLManager;
+
+@protocol CustomSDLScreen <NSObject>
+
+@required
+- (instancetype)initWithManager:(SDLManager *)sdlManager;
+- (void)showScreen;
+
+@end
+```
+```swift
+protocol CustomSDLScreen {
+    init(sdlManager: SDLManager)
+    func showScreen()
+}
+```
+~|
+!@
+
+@![android,javaSE,javaEE]
+```java
+    // TODO
+```
+!@
+
+@![javascript]
+```js
+    // TODO
+```
+!@
+
+### Screen Classes
+As previously mentioned most screens should all follow a shared protocol, such as the ```CustomSDLScreen``` protocol above, since screens will likely operate in a similar fashion. Another good practice for screen classes is to keep screen data in a view model. Doing so will add a layer of abstraction for exposing public properties and commands to the screen. 
+
+For the example below, the ```HomeScreen``` class will inherit the ```CustomSDLScreen``` protocol and will have a property of ```HomeDataViewModel```. The screen manager will change its text fields based on the view model's data. In addition, the home screen will also create a navigation button to open the ```SDLButtonScreen``` when pressed.
+
+@![iOS]
+|~
+```objc
+// HomeSDLScreen.h
+#import <Foundation/Foundation.h>
+#import "CustomSDLScreen.h"
+
+@class HomeDataViewModel;
+@class SDLManager;
+@class SDLScreenManager;
+@class SDLSoftButtonObject;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface HomeSDLScreen : NSObject<CustomSDLScreen>
+
+@end
+
+NS_ASSUME_NONNULL_END
+
+/************************************************************************/
+// HomeSDLScreen.m
+#import "HomeSDLScreen.h"
+
+#import "ButtonSDLScreen.h"
+#import "CustomSDLScreen.h"
+#import "HomeDataViewModel.h"
+#import "SmartDeviceLink.h"
+
+@interface HomeSDLScreen()
+
+@property (strong, nonatomic) SDLManager *sdlManager;
+// A button to navigate to the ButtonSDLScreen
+@property (strong, nonatomic) SDLSoftButtonObject *navigationButton;
+// An example of your data model that will feed data to the SDL screen's UI
+@property (strong, nonatomic) HomeDataViewModel *homeDataViewModel;
+
+@end
+
+@implementation HomeSDLScreen
+
+- (instancetype)initWithManager:(SDLManager *)sdlManager {
+    self = [super init];
+    if (!self) { return nil; }
+
+    _sdlManager = sdlManager;
+    // Set navigation button to an SDLSoftButtonObject that will display the Button SDL Screen
+    _navigationButton = [[SDLSoftButtonObject alloc] initWithName:@"ButtonSDLScreen" text:@"Button Screen" artwork:nil handler:^(SDLOnButtonPress * _Nullable buttonPress, SDLOnButtonEvent * _Nullable buttonEvent) {
+        if (buttonPress == nil) { return; }
+
+        ButtonSDLScreen *buttonScreen = [[ButtonSDLScreen alloc] initWithManager:sdlManager];
```

Comment:
Create the `ButtonSDLScreen` and hold it in the class, then call `[buttonScreen showScreen]`